mlflow/models/wheeled_model.py

```diff
@@ -213,12 +215,46 @@ def _download_wheels(cls, pip_requirements_path, dst_path):
         Args:
             pip_requirements_path: Path to requirements.txt in the model directory
             dst_path: Path to the directory where the wheels are to be downloaded
+            extra_envs: Extra environment variables to be passed to the subprocess.
         """
         if not os.path.exists(dst_path):
             os.makedirs(dst_path)
 
         pip_wheel_options = MLFLOW_WHEELED_MODEL_PIP_DOWNLOAD_OPTIONS.get()
 
+        allowed_options = [
+            "--only-binary=:all:",
+            "--only-binary=:none:",
+            "--no-binary=:all:",
+            "--no-binary=:none:",
+            "--prefer-binary",
+            "--no-build-isolation",
+            "--use-pep517",
+            "--check-build-dependencies",
+            "--ignore-requires-python",
+            "--no-deps",
+            "--no-verify",
+            "--pre",
+            "--require-hashes",
+            "--no-clean",
+        ]
+        all_options = pip_wheel_options.split(" ")
+        if any(option not in allowed_options for option in all_options):
```

Comment:
For group & group comparison, we could do this as well
```suggestion
        allowed_options = {
            "--only-binary=:all:",
            "--only-binary=:none:",
            "--no-binary=:all:",
            "--no-binary=:none:",
            "--prefer-binary",
            "--no-build-isolation",
            "--use-pep517",
            "--check-build-dependencies",
            "--ignore-requires-python",
            "--no-deps",
            "--no-verify",
            "--pre",
            "--require-hashes",
            "--no-clean",
        }
        all_options = set(pip_wheel_options.split(" "))
        if all_options.issubset(allowed_options):
```